src/Microsoft.AspNetCore.Http.Extensions/HeaderDictionaryTypeExtensions.cs

```diff
@@ -156,6 +156,7 @@ internal static void SetDate(this IHeaderDictionary headers, string name, DateTi
             { typeof(RangeHeaderValue), new Func<string, RangeHeaderValue>(value => { RangeHeaderValue result; return RangeHeaderValue.TryParse(value, out result) ? result : null; }) },
             { typeof(EntityTagHeaderValue), new Func<string, EntityTagHeaderValue>(value => { EntityTagHeaderValue result; return EntityTagHeaderValue.TryParse(value, out result) ? result : null; }) },
             { typeof(DateTimeOffset?), new Func<string, DateTimeOffset?>(value => { DateTimeOffset result; return HeaderUtilities.TryParseDate(value, out result) ? result : (DateTimeOffset?)null; }) },
+            { typeof(Uri), new Func<string, Uri>(value => { Uri result; return Uri.TryCreate(value, UriKind.RelativeOrAbsolute, out result) ? result : null; }) },
```

Comment:
out var result

Reply:
I wrote it in the same way as the `DateTimeOffset?` above and the `long?` below.
Do you want me to inline the var 'out' declaration?
I.e. `.TryParseDate(value, out var result)`

Reply:
yes